homeassistant/components/discovergy/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""The Discovergy integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+import logging
+
+import pydiscovergy
+from pydiscovergy.authentication import BasicAuth
+import pydiscovergy.error as discovergyError
+from pydiscovergy.models import Meter
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import APP_NAME, DOMAIN
+
+PLATFORMS = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class DiscovergyData:
+    """Discovergy data class to share meters and api client."""
+
+    api_client: pydiscovergy.Discovergy = field(default_factory=lambda: None)
```

Comment:
We don't need defaults if we pass the parameters when instantiating the dataclass.